### PR TITLE
DVC-4962

### PR DESCRIPTION
**Pull request by nikolasleblanc**
Reduces bundle size by ensuring the types package is treeshakable.

Also includes:

- local verdaccio install for local testing outside the monorepo and docs

### CODE REVIEW

File: lib/shared/types/tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
   "extends": "../../../tsconfig.base.json",
   "compilerOptions": {
-    "module": "CommonJS",
+    "target": "es2015",
```
Comment (ajwootto) [inline lib/shared/types/tsconfig.json:4]: can we target a more recent version for this?

File: lib/shared/types/project.json [hunk 2]
```diff
@@ -18,18 +18,25 @@
...
         "assets": [
-          "lib/shared/types/*.md"
-        ]
+          {
+            "glob": "*.md",
+            "input": "lib/shared/types",
+            "output": "."
+          }
+        ],
+        "skipTypeField": true
```
Comment (ajwootto) [inline lib/shared/types/project.json:39]: does it not work with that field?

File: lib/shared/types/tsconfig.json (hunk 1 above)
Comment (nikolasleblanc) [reply lib/shared/types/tsconfig.json:4]: Think so, I'll give it a shot.

File: lib/shared/types/project.json (hunk 2 above)
Comment (ajwootto) [reply lib/shared/types/project.json:39]: weird..

**Review by ajwootto — APPROVED** [review]

**Pull request merged**